app/forms.py

```diff
@@ -0,0 +1,143 @@
+        length_of_month = len(str(month))
+        if length_of_month > 2 or month == 0:
+            raise forms.ValidationError(u'Please enter a valid month')
+        return month
+
+    def clean_day(self):
+        day = self.cleaned_data['day']
+        length_of_day = len(str(day))
+        if length_of_day > 2 or day == 0:
+            raise forms.ValidationError(u'Please enter a valid day')
+        return day
+
+    def clean(self):
+        return self.cleaned_data
+
+
+class EditProfileForm(RegistrationForm):
+    alias = forms.CharField(
+        label=_("Display Name"),
+        required=False
+    )
+
+    def __init__(self, *args, **kwargs):
+        super(EditProfileForm, self).__init__(*args, **kwargs)
+        del self.fields['password1']
+        del self.fields['year']
+        del self.fields['month']
+        del self.fields['day']
+        del self.fields['username']
+        self.fields.keyOrder = [
+            'alias',
+        ]
+
+    def clean(self):
+        """
+        Check whether the user has edited the email field or not.
+        If user edit the email field,
+        we should check the email is present already or not.
+        Otherwise we can allow the data to update.
+        """
+        cleaned_data = super(EditProfileForm, self).clean()
+        return cleaned_data
+
+
+class ProfilePasswordChangeForm(forms.Form):
+    old_password = forms.CharField(widget=forms.PasswordInput())
+    new_password = forms.CharField(widget=forms.PasswordInput())
+    confirm_password = forms.CharField(widget=forms.PasswordInput())
+
+    def clean_old_password(self):
+        old_password = self.cleaned_data['old_password']
+        length_of_old_password = len(str(old_password))
+        if length_of_old_password > 4:
+            raise forms.ValidationError(u'Please enter a valid old password')
+        return old_password
+
+    def clean_new_password(self):
+        new_password = self.cleaned_data['new_password']
+        length_of_new_password = len(str(new_password))
+        if length_of_new_password > 4:
+            raise forms.ValidationError(u'New password should be \
+             maximum of 4 characters')
+        return new_password
+
+    def clean_confirm_password(self):
+        confirm_password = self.cleaned_data['confirm_password']
+        length_of_confirm_password = len(str(confirm_password))
+        if length_of_confirm_password > 4:
+            raise forms.ValidationError(u'Confirm new password should be \
+             maximum of 4 characters')
+        return confirm_password
+
+    def clean(self):
+        try:
+            if self.cleaned_data['new_password'] != \
+                    self.cleaned_data['confirm_password']:
+                raise forms.ValidationError(_('New passwords does not match'))
+            else:
+                return self.cleaned_data
+        except KeyError:
+            raise forms.ValidationError(_('Plase enter passwords \
+             in a valid format'))
```

Comment:
:point_up: please identify when this could happen or (of not or in doubt) rather remove it.